### ISSUE TITLE
Frequent failures when deploying functions (Unhandled error cleaning up build images)

### ISSUE THREAD

**Issue by manwithsteelnerves** [firebase/firebase-tools#3639]
###  Environment info
**firebase-tools:** 9.16.0

**Platform:**macOS

### [REQUIRED] Test case

Try deploying functions
```
firebase deploy --only functions
```
functions fail with "Unhandled error cleaning up build images."
```
 functions: cleaning up build files...
⚠  functions: Unhandled error cleaning up build images. This could result in a small monthly bill if not corrected. You can attempt to delete these images by redeploying or you can delete them manually at https://console.cloud.google.com/gcr/images/api-dev/us/gcf
```

### [REQUIRED] Expected behavior

Should not fail deploying this frequently.

### [REQUIRED] Actual behavior

For deploying all my functions(around 50) per environment, I need to redeploy nearly 6 times to finish it. It keeps failing frequently from past few days. 

 I'm currently having 4 environments and attempting nearly **30 times** to finish deploying!

**Comment by inlined** [comment]
Can you try deploying with `firebase deploy --only functions --debug` and including the error message?

FWIW, that error happens after your code has finished deploying. All your functions are deployed regardless of that error message.

**Comment by manwithsteelnerves** [comment]
Unfortunately, debug flag didn't help either. There is no relevant information.
So, I deleted manually gcf folders under container registry.

The above error log went off but still deploys kept failing.

Later... After 2 days of frustration...
I found that there are yellow ! marks in the cloud console for some of the functions. So, deleted all 50 functions and then it started deploying properly!

Now,

**What if i'm in production? I can't delete the functions and re-deploy. Can someone suggest a proper solution to this?**
